interview-questions.md

```diff
@@ -8,50 +8,135 @@ Then, research the question to expand on your answer. Even if you feel you have
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-Your answer:
+Your answer: Pair programming has a driver and a navigator. It is the job of the navigator to make all inputs into the code while the navigator instructs the driver on what to input. This allows for two sets of eyes to be on the document at all times and in some cases gives some ability to research and code at the same time. This cuts down on time, and correction errors as two people are much more likely to pick up on any small mistakes.
+
+Researched answer: There are different styles of pair programming. 
+
+The first is the driver-navigator style that we have been using. An advantage of this when done well is things get done quickly and accurately. Another take on this might be the driver describing what they plan to do next kind of like pseudo code then allowing the navigator to give input or decide the best way forward.
+
+There is the unstructured pair style. This is well unstructured and has no rules. Basically its just two people sitting at a computer working through a problem. Can be beneficial if both partners are at equal skill and are working through basic problemns. Can allow you to get a lot done in a short time. This is difficult to do remotely as switch off is too much of a burden.
+
+There is Strong style pair programming. The idea hear as that before anything goes to the computer it must pass through the partner's head first. This is described as being good for a very specific task that one partner may have a deeper knowledge on. Consists mostly of the more experienced programmer giving guidance for the less experienced programmer but you still want to switch at given time intervals. Consists of the stronger partner giving feedback on the other partners design decisions, and questions about problem solving. 
+
+The Ping-Pong method is a where you trade off rolls where one person writes code and the partner reviews it. Can create a dynamic where neither partner can dominate the work the way maybe navigation can do.
+
+The backseat navigator is more about a more experience navigator telling the driver every little detail about the solution in their mind and overseeing the implementation by the driver.
+
+The Tour Guide Style is where a more experience developer is explaining maybe a new framework to so somebody and is activily coding while fielding questions about why and how things work. The tour guide will type and explain concepts and the rationale behind decisions. This is kind of what the teachers are doing with us in the lectures.
+
+Distributed pairing is basically what we are doing. It brings people from many different locations and allows them to connect remotely and work on a project.
+
+On an aside I think the biggest benefit is that pair programming can increases focus over a long period. It may be distracting if you are in the zone but in the long run it feels like it could lead to more productivity.
```

Comment:
Wow!  Truly incredible answers in this entire section!  I can tell you did your research!  Excellent work 🎉